python/pyspark/serializers.py

```diff
@@ -298,7 +298,10 @@ def create_array(s, t):
             if t is not None and pa.types.is_timestamp(t):
                 s = _check_series_convert_timestamps_internal(s, self._timezone)
             try:
-                array = pa.Array.from_pandas(s, mask=mask, type=t, safe=self._safecheck)
+                if str(s.dtype) == 'category':
```

Comment:
is there another way to check if `category` type other than a string comparison?

Reply:
Also, I'd prefer if you just replace the original series `s` before the conversion `from_pandas`. like 's = s.astype...`

Reply:
Done